### PR TITLE
revert!: deprecate VDF crate

### PR DESCRIPTION
**Pull request by mrain**
closes: #649 

### This PR: 
Deprecates the VDF crate because the original implemented MinRoot is insecure. Also we didn't and won't implement the verification part.

### Key places to review:

 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [ ] Added relevant changelog entries to the `CHANGELOG.md` of touched crates.
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

File: vdf/src/hashchain.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+impl VDF for HashChain {
+    type PublicParameter = HashChainParam;
+    type Proof = [u8; 32];
+    type Input = [u8; 32];
+    type Output = [u8; 32];
+
+    fn setup<R: ark_std::rand::CryptoRng + ark_std::rand::RngCore>(
+        difficulty: u64,
+        prng: Option<&mut R>,
+    ) -> Result<Self::PublicParameter, VDFError> {
```
Comment (ss-es) [inline vdf/src/hashchain.rs:48]: Is it possible to make `setup` and `eval` infallible (not return a `Result`)? It looks like they can't fail in either implementation

not really a big deal: we can of course always handle the failure in HotShot

File: vdf/src/hashchain.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        Ok(HashChainParam { difficulty })
+    }
+
+    fn eval(
+        pp: &Self::PublicParameter,
+        input: &Self::Input,
+    ) -> Result<(Self::Output, Self::Proof), VDFError> {
+        let mut output = *input;
+        for _ in 0..pp.difficulty {
+            output = sha3::Keccak256::digest(&input).into();
```
Comment (ss-es) [inline vdf/src/hashchain.rs:58]: maybe I'm not reading this right, but should this be `digest(&output)`?

Comment (mrain) [reply vdf/src/hashchain.rs]: stupid me! fixed in [ffc603b](https://github.com/EspressoSystems/jellyfish/pull/683/commits/ffc603be8be262428dee883c35b5ac8520f9b619)

File: vdf/src/hashchain.rs (hunk 1 above)
Comment (mrain) [reply vdf/src/hashchain.rs:51]: I prefer to keep it. Error may occur when we are implementing Verifiable DF.

**Review by alxiong — COMMENTED** [review]
1. why are we keeping the minroot? afaik, no one, not even ourselves, is using it
2. why sh3 hashchain? is it being used or planned anywhere?

**Comment by mrain** [comment]
> 1. why are we keeping the minroot? afaik, no one, not even ourselves, is using it

No particular reason. Maybe we'll comeback to it sometime later (very not likely)

> 2. why sh3 hashchain? is it being used or planned anywhere?

I just picked a random hash function. Do you have other suggestions or would you prefer a somewhat generic implementation cc @ss-es

**Review by alxiong — APPROVED** [review]
I agree with the removal.

**Pull request merged**